aws/internal/keyvaluetags/update_tags_gen.go

```diff
@@ -2686,6 +2687,44 @@ func ResourcegroupsUpdateTags(conn *resourcegroups.ResourceGroups, identifier st
 	return nil
 }
 
+// Route53UpdateTags updates route53 service tags.
+// The identifier is typically the Amazon Resource Name (ARN), although
+// it may also be a different identifier depending on the service.
+func Route53UpdateTags(conn *route53.Route53, identifier string, resourceType string, oldTagsMap interface{}, newTagsMap interface{}) error {
+	oldTags := New(oldTagsMap)
+	newTags := New(newTagsMap)
+
+	if removedTags := oldTags.Removed(newTags); len(removedTags) > 0 {
+		input := &route53.ChangeTagsForResourceInput{
+			ResourceId:    aws.String(identifier),
+			ResourceType:  aws.String(resourceType),
+			RemoveTagKeys: aws.StringSlice(removedTags.Keys()),
+		}
+
+		_, err := conn.ChangeTagsForResource(input)
+
+		if err != nil {
+			return fmt.Errorf("error untagging resource (%s): %w", identifier, err)
+		}
+	}
+
+	if updatedTags := oldTags.Updated(newTags); len(updatedTags) > 0 {
+		input := &route53.ChangeTagsForResourceInput{
+			ResourceId:   aws.String(identifier),
+			ResourceType: aws.String(resourceType),
+			AddTags:      updatedTags.IgnoreAws().Route53Tags(),
+		}
+
+		_, err := conn.ChangeTagsForResource(input)
+
+		if err != nil {
+			return fmt.Errorf("error tagging resource (%s): %w", identifier, err)
+		}
+	}
```

Comment:
We likely can update the generator to combine the calls if the add and remove API calls are the same, but that feels like a later optimization we can do here.